art/attacks/evasion/lowprofool.py

```diff
@@ -0,0 +1,487 @@
+            > 'pearson' - Pearson correlation (string)
+            > function  - Custom function (callable object)
+            > vector    - Vector of feature importance (np.ndarray)
+        :param verbose: Verbose mode / Show progress bars.
+        """
+        super().__init__(estimator=classifier)
+
+        self.n_steps = n_steps
+        self.threshold = threshold
+        self.lambd = lambd
+        self.eta = eta
+        self.eta_decay = eta_decay
+        self.eta_min = eta_min
+        self.p = p
+        self.importance = importance
+        self.verbose = verbose
+
+        self._targeted = True
+        self.n_classes = self.estimator.nb_classes
+        self.n_features = self.estimator.input_shape[0]
+        self.importance_vec = None
+        if self.estimator.clip_values is None:
+            logger.warning(
+                "The `clip_values` attribute of the estimator is `None`, therefore this instance of LowProFool will by "
+                "default generate adversarial perturbations without clipping them."
+            )
+
+        self._check_params()
+
+        if isinstance(self.importance, np.ndarray):
+            self.importance_vec = self.importance
+
+        if eta_decay < 1 and eta_min > 0:
+            steps_before_min_eta_reached = np.ceil(np.log(eta_min / eta) / np.log(eta_decay))
+
+            if steps_before_min_eta_reached / self.n_steps < 0.8:
+                logger.warning(
+                    "The given combination of 'n_steps', 'eta', 'eta_decay' and 'eta_min' effectively sets learning "
+                    "rate to its minimal value after about {} steps out of all {}.".format(
+                        steps_before_min_eta_reached, self.n_steps
+                    )
+                )
+
+    @staticmethod
+    def logistic_loss(
+            true: np.ndarray,
+            predicted: np.ndarray
+    ) -> np.ndarray:
+        """
+        Logistic loss function. Produces average cross-entropy in the sample.
+
+        :param true: Ground truth (correct) labels for samples.
+        :param predicted: Predicted probabilities.
+        :return: Mean cross entropy.
+        """
+        p = np.array([1 - true, true])
+        q = np.array([1 - predicted, predicted])
+
+        return np.mean(
+            -1 * np.sum(p * np.log(q + 1e-13), axis=0), axis=1
+        ).reshape(-1, 1)
+
+    def __weighted_lp_norm(
+            self,
+            perturbations: np.ndarray
+    ) -> np.ndarray:
+        """
+        Lp-norm of perturbation vectors weighted by feature importance.
+
+        :param perturbations: Perturbations of samples towards being adversarial.
+        :return: Array with weighted Lp-norm of perturbations.
+        """
+        return self.lambd * np.linalg.norm(
+            self.importance_vec * perturbations, axis=1,
+            ord=(np.inf if self.p == "inf" else self.p)
+        ).reshape(-1, 1)
+
+    def __classification_loss_gradient(
```

Comment:
This method is just one line and used only once. Please inline the call to `self.estimator.loss_gradient` and replace the call to `__classification_loss_gradient`.